### PR TITLE
tests: benchmarks: move data_structure and cmsis_dsp tests to new ztest API

### PR DESCRIPTION
**Pull request by enjiamai**
Migrate following tests to new ztest API:

- tests/benchmarks/data_structure_perf/dlist_perf
- tests/benchmarks/data_structure_perf/rbtree_perf
- tests/benchmarks/cmsis_dsp/basicmath

### CODE REVIEW

File: tests/benchmarks/data_structure_perf/dlist_perf/src/dlist_perf.c [hunk 1]
```diff
@@ -219,7 +219,7 @@ void test_dlist_peak_head_tail(void)
  *
  * @ingroup lib_dlist_tests
  */
-void test_dlist_insert_and_remove(void)
+ZTEST(dlist_perf,  test_dlist_insert_and_remove)
```
Comment (stephanosio) [inline tests/benchmarks/data_structure_perf/dlist_perf/src/dlist_perf.c:222]: Nit: spurious space

```suggestion
ZTEST(dlist_perf, test_dlist_insert_and_remove)
```

File: tests/benchmarks/data_structure_perf/dlist_perf/src/dlist_perf.c [hunk 2]
```diff
@@ -254,14 +254,4 @@ void test_dlist_insert_and_remove(void)
 		&node[ARRAY_SIZE(node)/2 - 1]), "dlist can't remove a node in constant time");
 }
 
-/* ztest main entry */
-void test_main(void)
-{
-	ztest_test_suite(test_dlist,
-			ztest_unit_test(test_dlist_container),
```
Comment (stephanosio) [inline tests/benchmarks/data_structure_perf/dlist_perf/src/dlist_perf.c:261]: need to convert `test_dlist_container`

**Review by stephanosio — CHANGES_REQUESTED** [review]

File: tests/benchmarks/data_structure_perf/dlist_perf/src/dlist_perf.c (hunk 1 above)
Comment (enjiamai) [reply tests/benchmarks/data_structure_perf/dlist_perf/src/dlist_perf.c]: Thanks for catching this! Got it fixed.

File: tests/benchmarks/data_structure_perf/dlist_perf/src/dlist_perf.c (hunk 2 above)
Comment (enjiamai) [reply tests/benchmarks/data_structure_perf/dlist_perf/src/dlist_perf.c:261]: Oh, I missed this. Thanks for your reviewing! :)

**Review by stephanosio — APPROVED** [review]

**Review by aaronemassey — APPROVED** [review]

**Pull request merged**